airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -490,18 +499,45 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(
           firstRecordWaitTime,
           subsequentRecordWaitTime,
           OptionalInt.empty());
-
       final MssqlCdcConnectorMetadataInjector mssqlCdcConnectorMetadataInjector = MssqlCdcConnectorMetadataInjector.getInstance(emittedAt);
 
-      final Supplier<AutoCloseableIterator<AirbyteMessage>> incrementalIteratorSupplier = () -> handler.getIncrementalIterators(catalog,
+      // Determine if new stream(s) have been added to the catalog after initial sync of existing streams
+      final List<ConfiguredAirbyteStream> streamsToSnapshot = identifyStreamsToSnapshot(catalog, stateManager);
+      final ConfiguredAirbyteCatalog streamsToSnapshotCatalog = new ConfiguredAirbyteCatalog().withStreams(streamsToSnapshot);
+
+      final Supplier<AutoCloseableIterator<AirbyteMessage>> incrementalIteratorsSupplier = () -> handler.getIncrementalIterators(
+          catalog,
           new MssqlCdcSavedInfoFetcher(stateManager.getCdcStateManager().getCdcState()),
           new MssqlCdcStateHandler(stateManager),
           mssqlCdcConnectorMetadataInjector,
-          MssqlCdcHelper.getDebeziumProperties(database, catalog),
+          MssqlCdcHelper.getDebeziumProperties(database, catalog, false),
           DebeziumPropertiesManager.DebeziumConnectorType.RELATIONALDB,
-          emittedAt, true);
+          emittedAt,
+          true);
+
+      /*
+       * If the CDC state is null or there is no streams to snapshot, that means no stream has gone
```

Comment:
Is this correct? if no stream finished snapshot we want to snapshot all, not have an incremental sync, no?

Reply:
The naming is incorrect. It's actually snapshot for _new_streams.